pkg/api/steve/clusters/kubeconfig.go

```diff
@@ -74,22 +75,27 @@ func (k kubeconfigDownload) ServeHTTP(rw http.ResponseWriter, req *http.Request)
 }
 
 func (k kubeconfigDownload) ensureToken(userName string, req *http.Request) (string, error) {
+	defaultTokenTTL, err := tokens.GetKubeconfigDefaultTokenTTLInMilliSeconds()
+	if err != nil {
+		return "", fmt.Errorf("failed to get default token TTL: %w", err)
+	}
 
 	authToken, err := k.auth.TokenFromRequest(req)
 	if err != nil {
 		return "", err
 	}
+
 	tokenNamePrefix := fmt.Sprintf("kubeconfig-%s", userName)
 	input := user.TokenInput{
 		TokenName:     tokenNamePrefix,
 		Description:   "Kubeconfig token",
 		Kind:          "kubeconfig",
 		UserName:      userName,
 		AuthProvider:  authToken.AuthProvider,
-		TTL:           nil,
+		TTL:           defaultTokenTTL,
 		Randomize:     true,
 		UserPrincipal: authToken.UserPrincipal,
 	}
-	token, err := k.userMgr.EnsureToken(input)
-	return token, err
+
+	return k.userMgr.EnsureToken(input)
```

Comment:
I'm not as familiar with Steve yet, what path is this from ? Like in what case would this whole file get executed.

Reply:
Download KubeConfig from the UI, but only when MCM is disabled. [Relavent PR](https://github.com/rancher/rancher/pull/33811)

Reply:
@KevinJoiner can you make sure this is noted in the QA testing notes